File: C#/TopflytechCodec/TopflytechCodec/Entities/OneWireMessage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace TopflytechCodec.Entities
{
    public class OneWireMessage : Message
    {
        private DateTime date;
        private bool isIgnition = false;
        private string deviceId;
        private byte[] oneWireData;

        public string DeviceId
        {
            get { return deviceId; }
            set { deviceId = value; }
        }

        public byte[] OneWireData
        {
            get { return oneWireData; }
            set { oneWireData = value; }
        }

        public DateTime Date
        {
            get { return date; }
            set { date = value; }
        }

        public bool IsIgnition
        {
            get { return isIgnition; }
            set { isIgnition = value; }
        }
    }
}

File: Java/topflytechCodec/src/com/topflytech/codec/entities/OneWireMessage.java
package com.topflytech.codec.entities;
 
import java.util.Date;

public class OneWireMessage extends Message{
    private Date date;
    private boolean isIgnition = false;
    private String deviceId;
    private byte[] oneWireData;

    public String getDeviceId() {
        return deviceId;
    }

    public void setDeviceId(String deviceId) {
        this.deviceId = deviceId;
    }

    public byte[] getOneWireData() {
        return oneWireData;
    }

    public void setOneWireData(byte[] oneWireData) {
        this.oneWireData = oneWireData;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    public boolean isIgnition() {
        return isIgnition;
    }

    public void setIsIgnition(boolean ignition) {
        isIgnition = ignition;
    }
}
